x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch

```diff
@@ -0,0 +1,191 @@
+Clang16 will not allow implicit function declarations and implicit integers etc.
+This patch overhauls the source code for modern C.
+
+Bug: https://bugs.gentoo.org/875704
+Upstream PR: https://github.com/oshazard/gtkdialog/pull/81
+
+Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>
+
+--- a/src/actions.c
++++ b/src/actions.c
+@@ -36,6 +36,7 @@
+ #include "attributes.h"
+ #include "variables.h"
+ #include "tag_attributes.h"
++#include "gtkdialog_parser.h"
+ 
+ extern gchar *option_include_file;
+ 
+--- a/src/automaton.c
++++ b/src/automaton.c
+@@ -35,6 +35,7 @@
+ */
+ 
+ #include <gtk/gtk.h>
++#include <glib/gprintf.h>
+ 
+ #include "config.h"
+ #include "gtkdialog.h"
+@@ -87,6 +88,8 @@
+ #undef TOOLTIPS
+ 
+ extern gboolean option_no_warning;
++extern void push_widget(GtkWidget * widget, int widgettype);
++int instruction_execute(instruction command);
+ 
+ instruction *program = NULL;
+ int instruction_counter = 0;		/* The first available memory cell */
+@@ -323,8 +326,7 @@ void print_command(instruction command)
+     fflush(stdout);
+ }
+ 
+-void
+-print_program()
++void print_program()
+ {
+ 	gint pc;
+ 	instruction command;
+@@ -580,8 +582,7 @@ finalize:
+ }
+ 
+ 
+-int 
+-instruction_execute(instruction command)
++int instruction_execute(instruction command)
+ {
+ 	GList *element;
+ 	token Token;
+@@ -911,9 +912,11 @@ gboolean widget_moved(GtkWidget *widget,
+ 		//gtk_widget_set_usize(window, 
+ 		//		configure->width -20,
+ 		//		configure->height);
++#if HAVE_GTK==2
```

Comment:
gtk_widget_set_uposition is defined in `/usr/include/gtk-2.0/gtk-gtkwidget.h`. 
If gtkdialog is configured to use gtk3 (what we hardcode in the ebulid) we can not assume it is there and include that file. 
That is why I put that pp-if there. But does this work this way? HAVE_GTK is defined in configure.

Reply:
I think this looks fine.